src/main/benchmark.c

```diff
@@ -112,6 +111,13 @@ run_benchmark(args_t* args)
 		ret = initialize_histograms(&data, args, &start_time, &start_timespec);
 	}
 
+	if (ret == 0) {
+		for (uint32_t i = 0; i < data.stages.n_stages && ret == 0; i++) {
+			ret = obj_spec_bin_name_compatible(&data.stages.stages[i].obj_spec,
```

Comment:
That's right. Since each object-spec in each stage could potentially be of different sizes (have a different number of bins), we go through and make sure the bin name isn't too long for any of the object specs. If ret ever becomes something other than 0, it skips the rest of initialization and proceeds immediately to cleanup.